src/infer_request.cc

```diff
@@ -1005,21 +1005,54 @@ InferenceRequest::Normalize()
       (original_inputs_.size() < model_raw_->RequiredInputCount())) {
     // If no input is marked as optional, then use exact match error message
     // for consistency / backward compatibility
+    std::string missing_required_input_string = "[";
```

Comment:
This Normalize() function is already quite long. What do you think about refactoring this block into a helper function?
```cpp
InferenceRequest::Normalize() {
  ...
  // Make sure that the request is providing the number of inputs
  // as is expected by the model.
  RETURN_IF_ERROR(ValidateRequestInputs(...));

  ...
}

Status ValidateRequestInputs(...) {
    if ((original_inputs_.size() > (size_t)model_config.input_size()) ||
      (original_inputs_.size() < model_raw_->RequiredInputCount())) {
      ...
        if ((size_t)model_config.input_size() == model_raw_->RequiredInputCount()) { 
          return Status(...)
       } else {
         return Status(...)
       }
    }

    return Status::Success;
}
```

Reply:
Yes this looks neat.
Will update the PR with a quick round of test